### PR TITLE
feat: use custom widget for living documentation

### LINKED ISSUE #496: Living-documentation: Use custom widgets to dynamically display information
### What

The living-documentation currently dynamically shows DNS resolution information. It is implemented using a lambda which is periodically (every minute) invoked. The user need still needs to manually reload the dashboard as the text widget is not automatically updated.

We can instead use custom widgets to dynamically fetch data from lambda only when the cloudwatch dashboard is invoked.

### Why

Since the [introduction of custom widgets](https://aws.amazon.com/about-aws/whats-new/2021/08/custom-widgets-amazon-cloudwatch-dashboards/) there is another way to dynamically fetch information by directly invoking lambda functions. 
This allows to use different lambda functions for different purposes (e.g. one lambda to fetch security and one lambda to fetch billing information), adhering to the single responsibility principle. Also the information would only be fetched when the user wants to see it and not every minute (although this is still within the free tier).

See [documentation](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/add_custom_widget_dashboard.html)

### PR DESCRIPTION
**Pull request by peermuellerxw**
Closes #496

### CODE REVIEW

File: tests/living_documentation_test.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import boto3
-from retrying import retry
+import pytest
+import botocore
 
 cw = boto3.client('cloudwatch')
 lambda_client = boto3.client('lambda')
 
 
 def test_superwerker_dashboard():
```
Comment (aperov9) [inline tests/living_documentation_test.py:9]: we are not testing if the contents are properly displayed?

File: cdk/src/stacks/living-documentation.ts [hunk 2]
```diff
@@ -37,6 +35,20 @@ export class LivingDocumentationStack extends NestedStack {
...
     (dashboardGeneratorFunction.node.defaultChild as lambda.CfnFunction).overrideLogicalId('DashboardGeneratorFunction');
 
+    const superwerkerDashboard = new Dashboard(this, 'SuperwerkerDashboard', {
+      dashboardName: 'Superwerker-LivingDocumentation',
+    });
+
+    superwerkerDashboard.addWidgets(
+      new CustomWidget({
+        title: '',
+        width: 20,
```
Comment (aperov9) [inline cdk/src/stacks/living-documentation.ts:45]: Is there an option to use full width?

File: cdk/src/functions/living-docs-dashboard-generator.ts [hunk 3]
```diff
@@ -1,38 +1,55 @@
-import { CloudWatchClient, PutDashboardCommand } from '@aws-sdk/client-cloudwatch';
+import { CloudWatchClient, DeleteDashboardsCommand } from '@aws-sdk/client-cloudwatch';
 import { SSMClient, GetParameterCommand } from '@aws-sdk/client-ssm';
 import endent from 'endent';
 
 const ssmClient = new SSMClient();
 const cloudwatchClient = new CloudWatchClient({});
 
-export async function handler(_event: any, _context: any) {
+const DOCS = `
```
Comment (aperov9) [inline cdk/src/functions/living-docs-dashboard-generator.ts:8]: this is shown while the lambda is fetching the contents? I could not see it inside

Comment (peermuellerxw) [reply cdk/src/functions/living-docs-dashboard-generator.ts:8]: This is shown in the UI when you want to add a new custom widget.

File: tests/living_documentation_test.py (hunk 1 above)
Comment (peermuellerxw) [reply tests/living_documentation_test.py]: Good point. I added another test.

File: cdk/src/stacks/living-documentation.ts (hunk 2 above)
Comment (peermuellerxw) [reply cdk/src/stacks/living-documentation.ts:45]: There are 24 columns. So the maximum is 24. Let's try this.

**Pull request merged**